File: GoogleTagManagerDemo/GoogleTagManagerDemo/ViewController.m
//
//  ViewController.m
//  GoogleTagManagerDemo
//
//  Created by xuzepei on 16/3/1.
//  Copyright © 2016年 xuzepei. All rights reserved.
//

#import "ViewController.h"
#import "TAGDataLayer.h"
#import "TAGManager.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)clickedButton:(id)sender {
    
    //Push data will trigger the function if any custom tag's rule(s) evalute to true
    TAGDataLayer *dataLayer = [TAGManager instance].dataLayer;
    [dataLayer push:@{@"event": @"openScreen", @"screenName": @"Home Screen"}];
}

@end

File: GoogleTagManagerSwiftDemo/GoogleTagManagerSwiftDemo/ViewController.swift
//
//  ViewController.swift
//  GoogleTagManagerSwiftDemo
//
//  Created by xuzepei on 16/3/3.
//  Copyright © 2016年 xuzepei. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var dataLayer: TAGDataLayer = TAGManager.instance().dataLayer

    @IBOutlet weak var myButton: UIButton!
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        dataLayer.push(["event": "OpenScreen", "screenName": "Home Screen"])
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func clickedButton(sender: AnyObject) {
        
        dataLayer.push(["event": "ButtonPressed", "eventAction": "Test Event Action"])
    }

}

